### PR TITLE
Add example for callbacks and lift requirement for tagging inline

### PR DESCRIPTION
**Pull request by dalg24**
Example was long due and not requesting to tag callbacks makes the most common case easier and allows for generic lambdas.

### CODE REVIEW

File: examples/callback/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#if(Kokkos_ENABLE_CUDA)
+#  add_executable(ArborX_CudaAccessTraits.exe example_cuda_access_traits.cpp)
+#  target_link_libraries(ArborX_CudaAccessTraits.exe ${ArborX_TARGET})
+#  add_test(NAME ArborX_CudaAccessTraits_Example COMMAND ./ArborX_CudaAccessTraits.exe)
+#endif()
```
Comment (aprokop) [inline examples/callback/CMakeLists.txt:5]: ???

File: examples/callback/example_callback.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          MemorySpace{},
+          Kokkos::View<ArborX::Point *, Kokkos::HostSpace,
+                       Kokkos::MemoryUnmanaged>(points.data(), points.size()))};
+
+  {
+    Kokkos::View<int *, MemorySpace> values("values", 0);
+    Kokkos::View<int *, MemorySpace> offsets("offsets", 0);
+    bvh.query(ExecutionSpace{}, FirstOctant{}, PrintfCallback{}, values,
+              offsets);
+#ifndef __NVCC__
```
Comment (aprokop) [inline examples/callback/example_callback.cpp:109]: This seems to be unusual check. We usually protect with something like `__CUDA__`, but I guess you checked that it works with Clang+Cuda?

File: examples/callback/example_callback.cpp [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+  // EXPERIMENTAL
+  // TODO replace with BVH::query(ExecutionSpace, Predicates, Callback) when
+  // new overload is added
+  Kokkos::View<int, ExecutionSpace, Kokkos::MemoryTraits<Kokkos::Atomic>> c(
+      "counter");
+
+  ArborX::Details::traverse(
+      ExecutionSpace{}, bvh, FirstOctant{},
+      KOKKOS_LAMBDA(int i, int j) { printf("%d %d %d\n", ++c(), i, j); });
```
Comment (aprokop) [inline examples/callback/example_callback.cpp:146]: Maybe put the block in `{}` too.

File: test/tstCallbacks.cpp [hunk 4]
```diff
@@ -51,6 +51,18 @@ struct NearestPredicateCallbackMissingTag
   }
 };
 
+struct Wrong
+{
+};
+
+struct SpatialPredicateCallbackDoesNotTakeRightFirstArgument
```
Comment (aprokop) [inline test/tstCallbacks.cpp:58]: Right -> Correct

File: examples/callback/CMakeLists.txt (hunk 1 above)
Comment (dalg24) [reply examples/callback/CMakeLists.txt:5]: Not intentional

File: examples/callback/example_callback.cpp (hunk 2 above)
Comment (dalg24) [reply examples/callback/example_callback.cpp:109]: Yes it is only a limitation with NVCC

File: examples/callback/example_callback.cpp (hunk 3 above)
Comment (dalg24) [reply examples/callback/example_callback.cpp:146]: Sure

File: test/tstCallbacks.cpp (hunk 4 above)
Comment (dalg24) [reply test/tstCallbacks.cpp:58]: K

**Pull request merged**